sky/jobs/state.py

```diff
@@ -443,40 +443,44 @@ def set_failed(
     logger.info(failure_reason)
 
 
-def set_cancelling(job_id: int, callback_func: CallbackType):
-    """Set tasks in the job as cancelling, if they are in non-terminal states.
-
-    task_id is not needed, because we expect the job should be cancelled
-    as a whole, and we should not cancel a single task.
-    """
```

Comment:
I think we should reconsider the cancellation strategy. Since each task writes its output to the data bucket:

1. We should prevent new tasks from starting (previous behavior, which I agree)
2. But running tasks should be allowed to complete because:
   - Their execution might be valid and outputs useful
   - Results can be read from bucket when resuming the workflow
   - Cancelling running threads adds unnecessary complexity
3. Cancelling running tasks wastes resources and loses potentially valid results

Thoughts?

Reply:
Whether the output is useful is determined by the task. Not all task's intermediate result is useful. For example, for the graph A -> B, A -> C, if B aborts and the bug on B task stems from A, and C is running now, we should stop C as it will abort sooner or later.

I think both cases are possible and it is really hard to tell which case happens more commonly, so both solutions makes sense. The problem boils down to which behaviour we should choose as a default behaviour.

> Cancelling running threads adds unnecessary complexity

Considering this, lets just do what is easier to implement. Please left a TODO in the code and we can discuss more when we merge into the master.